core/unit_test/TestGraph.hpp

```diff
@@ -739,4 +739,61 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph), end_of_submit_control_flow) {
             value_A + 2 * value_B + value_C + value_D + value_E + value_F);
 }
 
+template <typename Exec>
+struct GraphNodeTypes {
+  // Type of a kernel node built using a Kokkos parallel construct.
+  using kernel_t =
+      Kokkos::Impl::GraphNodeKernelImpl<Exec, Kokkos::RangePolicy<Exec>,
+                                        CountTestFunctor<Exec>,
+                                        Kokkos::ParallelForTag>;
+
+  // Type of an aggregate node.
+  using aggregate_t = typename Kokkos::Impl::GraphImpl<Exec>::aggregate_impl_t;
+};
```

Comment:
I would get rid of that class template and inline the definitions below.  I don't think it helps.
(Not blocking though)

Reply:
It does not really help now, but in #7552 I'll add a new node type. And therefore I'll add a `is_graph_then` test. So this helper will become more useful later :wink: